ui/apps/platform/src/Containers/Collections/CollectionAttacher.tsx

```diff
@@ -1,7 +1,95 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
+import { Alert, Button, debounce, Flex, SearchInput, Truncate } from '@patternfly/react-core';
 
-function CollectionAttacher() {
-    return <></>;
+import BacklogListSelector from 'Components/PatternFly/BacklogListSelector';
+import { CollectionResponse } from 'services/CollectionsService';
+import useEmbeddedCollections from './hooks/useEmbeddedCollections';
+
+const selectorListCells = [
+    {
+        name: 'Name',
+        render: ({ name }) => (
+            <Button variant="link" className="pf-u-pl-0" isInline>
+                {name}
+            </Button>
+        ),
+    },
+    {
+        name: 'Description',
+        render: ({ description }) => <Truncate content={description} />,
+    },
+];
+
+export type CollectionAttacherProps = {
+    initialEmbeddedCollections: CollectionResponse[];
+    onSelectionChange: (collections: CollectionResponse[]) => void;
+};
+
+function compareNameLowercase(search: string): (item: { name: string }) => boolean {
+    return ({ name }) => name.toLowerCase().includes(search.toLowerCase());
+}
+
+function CollectionAttacher({
+    initialEmbeddedCollections,
+    onSelectionChange,
+}: CollectionAttacherProps) {
+    const [search, setSearch] = useState('');
+    const embedded = useEmbeddedCollections(initialEmbeddedCollections);
+    const { attached, detached, attach, detach, hasMore, fetchMore, onSearch } = embedded;
+    const { isFetchingMore, fetchMoreError } = embedded;
+
+    const onSearchInputChange = useMemo(
+        () =>
+            debounce((value: string) => {
+                setSearch(value);
+                onSearch(value);
+            }, 800),
+        [onSearch]
+    );
+
+    const selectedOptions = attached.filter(compareNameLowercase(search));
+    const deselectedOptions = detached.filter(compareNameLowercase(search));
+
+    return (
+        <Flex direction={{ default: 'column' }} spaceItems={{ default: 'spaceItemsXl' }}>
+            <SearchInput
+                aria-label="Search by name"
+                placeholder="Search by name"
+                value={search}
+                onChange={onSearchInputChange}
+            />
+            <BacklogListSelector
+                selectedOptions={selectedOptions}
+                deselectedOptions={deselectedOptions}
+                onSelectItem={({ id }) => attach(id)}
+                onDeselectItem={({ id }) => detach(id)}
```

Comment:
An observation, not a change request, that change in terminology from attach/detach to select/deselect is what I would expect for a reusable component, what do you think are pro and con:
* pro: how likely to reuse in a context where application level is other than attach/detach
* con: slight substitution by reader and possible effect on Find in Files

Reply:
Hmm this is a good question. My gut feeling is that the default terminology for using this in other contexts would be "select/deselect". This component is similar to the [dual list selector PF](https://www.patternfly.org/v4/components/dual-list-selector) component which uses terms that are fairly close. I think the attach/detach are specific, but not exclusive, to the semantics behind Collections.

That said it doesn't feel great having a disconnect between the props and the only use case. (The `useSelectToggle()` hook suffers from this mismatch as well.) I'm OK with changing the terminology or leaving as-is. Do you think people are more likely to Find in Files or Ctrl+Click to navigate around this code in the future?

Reply:
Your choice. You caught me: I often use Find in Files by mental muscle memory from less capable editors.